### PR TITLE
Replace cover-function-only by cover-include-pattern

### PR DESCRIPTION
**Pull request by peterschrammel**
Companion: https://github.com/diffblue/test-gen/pull/1031

### CODE REVIEW

**Comment by thk123** [comment]
Could I ask for some explanation either in the commit or on the PR as to what this is intending to do?

File: src/goto-instrument/cover.cpp
Comment (mgudemann) [inline src/goto-instrument/cover.cpp]: for `regex` we should escape `.`, else there could be a problem with correct matching, may also be required for other characters that can be part of a function name, maybe '<', '>' ?

**Review by mgudemann — CHANGES_REQUESTED** [review]

**Comment by peterschrammel** [comment]
@thk123, amended the commit message.

**Review by thk123 — CHANGES_REQUESTED** [review]
Somewhat incredibly, there are _no_ tests in cbmc/regression using the `--cover-function-only` flag. Could you add some in the cbmc-cover directory:

- A default entry point
- A default entry point specified by the `--function` flag
- A function specified with the `--function` flag using the minimal format: `--function Class.functionName`
- A function specified with the `--function` flag using the complex format that includes types etc (I'm not an expert on how this works, it might be appropriate to test static methods, non-static methods, methods that take and return arrays, generic methods)

As a first step you should probably bump the test-gen submodule  (which does have `--cover-function-only` tests) and check they pass with this change.

File: src/goto-instrument/cover.cpp
Comment (thk123) [reply src/goto-instrument/cover.cpp]: This appears to be a robust way of doing this: https://stackoverflow.com/a/40195721/958004

**Comment by thk123** [comment]
@peterschrammel thanks for the commit message - makes sense :+1:

**Comment by peterschrammel** [comment]
These options have never been made available in CBMC's CLI (as many others...). There are tests in test-gen, but not as systematic as you suggest...

**Comment by thk123** [comment]
@peterschrammel Ah that makes sense - well OK I won't block on that, but I think a PR to test-gen that brings this in should probably test complex format style of specifying functions since I believe this is what the platform uses (and would have caught the regex escape problem)

**Review by mgudemann — APPROVED** [review]
LGTM

**Comment by thk123** [comment]
Will approve once test-gen#1031 is working as expected.

**Comment by tautschnig** [comment]
@thk123 What's the state of TG on this one?

**Comment by thk123** [comment]
Sorry been away and then busy - current state of TG is failing however...

**Comment by peterschrammel** [comment]
Closing because obsolete (subsumed by https://github.com/diffblue/cbmc/pull/1419).

**Pull request closed without merging**
